### PR TITLE
Add some function nullable info for CallOperator

### PR DESCRIPTION
**Pull request by Youngwb**

### CODE REVIEW

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CallOperator.java [hunk 1]
```diff
@@ -109,7 +120,18 @@ public void setChild(int index, ScalarOperator child) {
...
+        if (fn == null) {
+            return true;
+        }
+        // check if fn always return non null
+        if (!fn.isNullable()) {
+            return false;
+        }
+        // check children nullable
+        if (nullableSameWithChildrenFunctions.contains(fnName)) {
+            return arguments.stream().anyMatch(ScalarOperator::isNullable);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CallOperator.java:132]: I think you can't use child‘s nullable info, our nullable info is not accurate now

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CallOperator.java]: done

**Review by HangyuanLiu — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CallOperator.java [hunk 2]
```diff
@@ -110,24 +107,9 @@ public void setChild(int index, ScalarOperator child) {
...
-                    .add(FunctionSet.BITMAP_UNION_INT)
-                    .add(FunctionSet.BITMAP_UNION_COUNT)
-                    .add(FunctionSet.BITMAP_COUNT)
-                    .build();
-
-    // TODO(kks): improve this
     @Override
     public boolean isNullable() {
-        return !AlwaysReturnNonNullableFunctions.contains(fnName);
+        return fn == null || fn.isNullable();
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CallOperator.java:112]: needn't change.

File: fe/fe-core/src/main/java/com/starrocks/analysis/FunctionCallExpr.java [hunk 3]
```diff
@@ -828,7 +835,14 @@ public boolean hasNullableChild() {
 
     // TODO(kks): improve this
     public boolean isNullable() {
-        return !CallOperator.AlwaysReturnNonNullableFunctions.contains(fnName.getFunction());
+        if (FunctionSet.alwaysReturnNonNullableFunctions.contains(fnName.getFunction())) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/analysis/FunctionCallExpr.java:838]: use function nullable

**Review by kangkaisen — DISMISSED** [review]

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**